docker/autoscaler/run_autoscaler_with_retries.py

```diff
@@ -0,0 +1,26 @@
+import os
```

Comment:
I think this would be helpful when user run this in non-container environment? If user run this in kubernetes, I feel we can leverage pod restart to achieve the same goal

Reply:
Yes, I will try that before merging the PR :)

Reply:
Repeating sentiment to avoid crash-looping when possible.

Reply:
Btw, Dmitri, do you remember where the crash-looping was coming from? We might be able to avoid it by doing re-tries at a lower level (e.g. when connecting to Redis), that might lead to a more robust solution overall.

Reply:
Right, the retries are for connecting to the head node. Waiting for the Ray cluster to be ready in a more targeted / principled way sounds better.